packages/base/adapters/solana/web3js/providers/WalletConnectProvider.ts

```diff
@@ -217,7 +217,20 @@ export class WalletConnectProvider extends ProviderEventEmitter implements Provi
   }
 
   private get sessionChains() {
-    return this.session?.namespaces['solana']?.chains || []
+    const solanaNamespace = this.session?.namespaces['solana']
+
+    if (!solanaNamespace) {
+      return []
+    }
+
+    const chains = solanaNamespace.chains || []
+    const accountsChains = solanaNamespace.accounts.map(account => {
+      const [chainNamespace, chainId] = account.split(':')
```

Comment:
Why we are splitting them and merging again? Does this necessary?

Reply:
Ahh okay it's CAIP address, not CAIP network id got it

Reply:
we does it because it is a caip address, so it will be like `solana:chain_id:address`